lesson4/code-examples/examples/exercise.js

```diff
@@ -0,0 +1,217 @@
+// Write function to which will get getCountryByCode('DZ');     Use array functions
+
+function getCountryByCode(code) {
+  const countries = [
+    { name: "Afghanistan", code: "AF" },
+    { name: "Åland Islands", code: "AX" },
+    { name: "Albania", code: "AL" },
+    { name: "Algeria", code: "DZ" }
+  ];
+  const result = countries.filter(
+    country => country.code === code.toUpperCase()
+  );
+  return result.length > 0 ? result[0].name : "Country not found!";
+}
+
+function countryFinder() {
+  const countryCode = document.getElementById("task-one__input").value;
+  const result = getCountryByCode(countryCode);
+  document.querySelector(".task-one-result").textContent = result;
+}
+
+// Rewrite function using es7 await / async
+
+// var fetch = require('node-fetch');
+
+async function getCountries() {
+  const res = await fetch(
+    "https://restcountries.eu/rest/v2/name/Indonesia?fullText=true"
+  );
+  const json = await res.json();
+  let code = await json[0].alpha2Code;
+  const res2 = await fetch("http://country.io/phone.json");
+  const json2 = await res2.json();
+  return json2[code];
+}
+const result2 = getCountries().then(data => {
+  document.querySelector(".task-two-result").textContent = data;
+});
+
+// Our objective: get the total score of force users only. Let’s do it step by step!
+
+var personnel = [
+  {
+    id: 5,
+    name: "Luke Skywalker",
+    pilotingScore: 98,
+    shootingScore: 56,
+    isForceUser: true
+  },
+  {
+    id: 82,
+    name: "Sabine Wren",
+    pilotingScore: 73,
+    shootingScore: 99,
+    isForceUser: false
+  },
+  {
+    id: 22,
+    name: "Zeb Orellios",
+    pilotingScore: 20,
+    shootingScore: 59,
+    isForceUser: false
+  },
+  {
+    id: 15,
+    name: "Ezra Bridger",
+    pilotingScore: 43,
+    shootingScore: 67,
+    isForceUser: true
+  },
+  {
+    id: 11,
+    name: "Caleb Dume",
+    pilotingScore: 71,
+    shootingScore: 85,
+    isForceUser: true
+  }
+];
+
+function getTotalScore() {
+  const scores = personnel.map(person => {
```

Comment:
It would be better to use `filter` to filter out only force users. Then you could easily reduce that array calculating total scores without use of `parseInt()`